engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -0,0 +1,2532 @@
+        return null;
+        // throw new RuntimeException(e.toString());
+      }
+    } else {
+      throw new RuntimeException( "The function call getDayNumber requires 2 arguments." );
+    }
+  }
+
+  public static Object isWorkingDay( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    if ( ArgList.length == 1 ) {
+      try {
+        if ( isNull( ArgList[0] ) ) {
+          return null;
+        } else if ( isUndefined( ArgList[0] ) ) {
+          return undefinedValue;
+        } else {
+          java.util.Date dIn = (java.util.Date) ArgList[0];
+          Calendar startDate = Calendar.getInstance();
+          startDate.setTime( dIn );
+          if ( startDate.get( Calendar.DAY_OF_WEEK ) != Calendar.SATURDAY
+            && startDate.get( Calendar.DAY_OF_WEEK ) != Calendar.SUNDAY ) {
+            return Boolean.TRUE;
+          }
+          return Boolean.FALSE;
+        }
+      } catch ( Exception e ) {
+        return null;
+      }
+    } else {
+      throw new RuntimeException( "The function call isWorkingDay requires 1 argument." );
+    }
+  }
+
+  @SuppressWarnings( "unused" )
+  public static Object fireToDB( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+
+    Object oRC = new Object();
+    if ( ArgList.length == 2 ) {
+      try {
+        Object scmO = actualObject.get( "_step_" );
+        Script scm = (Script) scmO;
+        String strDBName = (String) ArgList[0];
+        String strSQL = (String) ArgList[1];
+        DatabaseMeta ci = DatabaseMeta.findDatabase( scm.getTransMeta().getDatabases(), strDBName );
+        if ( ci == null ) {
+          throw new RuntimeException( "Database connection not found: " + strDBName );
+        }
+        ci.shareVariablesWith( scm );
+
+        Database db = new Database( scm, ci );
+        db.setQueryLimit( 0 );
+        try {
+          if ( scm.getTransMeta().isUsingUniqueConnections() ) {
+            synchronized ( scm.getTrans() ) {
+              db.connect( scm.getTrans().getTransactionId(), scm.getPartitionID() );
+            }
+          } else {
+            db.connect( scm.getPartitionID() );
+          }
+
+          ResultSet rs = db.openQuery( strSQL );
+          ResultSetMetaData resultSetMetaData = rs.getMetaData();
+          int columnCount = resultSetMetaData.getColumnCount();
+          if ( rs != null ) {
+            List<Object[]> list = new ArrayList<Object[]>();
+            while ( rs.next() ) {
+              Object[] objRow = new Object[columnCount];
+              for ( int i = 0; i < columnCount; i++ ) {
+                objRow[i] = rs.getObject( i + 1 );
+              }
+              list.add( objRow );
+            }
+            Object[][] resultArr = new Object[list.size()][];
+            list.toArray( resultArr );
+            db.disconnect();
+            return resultArr;
+          }
+        } catch ( Exception er ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 532.